flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImplTest.java

```diff
@@ -135,154 +142,165 @@ public void testAbortOldAndStartNewCheckpoint() throws Exception {
-                                        callFinish(writer);
-                                    } catch (IllegalArgumentException e) {
-                                        // ignore here - should rethrow in close
-                                    }
-                                }));
-    }
-
-    @Test(expected = TestException.class)
-    public void testRethrowOnNextCall() throws Exception {
+        assertThat(buffer.isRecycled()).isTrue();
+    }
+
+    @Test
+    void testNoAddDataAfterFinished() {
+        assertThatThrownBy(
+                        () ->
+                                runWithSyncWorker(
+                                        writer -> {
+                                            callStart(writer);
+                                            callFinish(writer);
+                                            callAddInputData(writer);
+                                        }))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void testAddDataNotStarted() {
+        assertThatThrownBy(
+                        () ->
+                                runWithSyncWorker(
+                                        (Consumer<ChannelStateWriter>) this::callAddInputData))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void testFinishNotStarted() {
+        assertThatThrownBy(() -> runWithSyncWorker(this::callFinish))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void testRethrowOnClose() {
+        assertThatThrownBy(
+                        () ->
+                                runWithSyncWorker(
+                                        writer -> {
+                                            try {
+                                                callFinish(writer);
+                                            } catch (IllegalArgumentException e) {
+                                                // ignore here - should rethrow in
+                                                // close
+                                            }
+                                        }))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void testRethrowOnNextCall() {
         SyncChannelStateWriteRequestExecutor worker = new SyncChannelStateWriteRequestExecutor();
         ChannelStateWriterImpl writer =
                 new ChannelStateWriterImpl(TASK_NAME, new ConcurrentHashMap<>(), worker, 5);
-        writer.open();
-        worker.setThrown(new TestException());
-        unwrappingError(TestException.class, () -> callStart(writer));
+        assertThatThrownBy(
+                        () -> {
+                            writer.open();
+                            worker.setThrown(new TestException());
+                            callStart(writer);
+                        })
+                .hasCauseInstanceOf(TestException.class);
```

Comment:
```suggestion
        writer.open();
        worker.setThrown(new TestException());
        assertThatThrownBy(() -> callStart(writer)).hasCauseInstanceOf(TestException.class);
```
More-specific `assertThatThrownBy` help here as well.